packages/contracts-ops/contracts/scripts/UpgradeCallBatches.s.sol

```diff
@@ -0,0 +1,236 @@
+            generateGovernanceCalls(domain);
+        }
+        writeCallBatch(recovery);
+    }
+
+    function setUp() internal {
+        __Config_initialize(configFile);
+        string memory outputFile = "upgradeActions.json";
+        __CallBatch_initialize(
+            localDomainName,
+            getDomainNumber(localDomainName),
+            outputFile,
+            true
+        );
+        networksArray = getNetworks();
+    }
+
+    function loadBeacons(string memory domain) internal {
+        bridgeTokenBeacon = address(bridgeTokenUpgrade(domain).beacon);
+        tokenRegistryBeacon = address(tokenRegistryUpgrade(domain).beacon);
+        bridgeRouterBeacon = address(bridgeRouterUpgrade(domain).beacon);
+        governanceRouterBeacon = address(
+            governanceRouterUpgrade(domain).beacon
+        );
+        governanceRouterProxy = address(governanceRouterUpgrade(domain).proxy);
+        for (uint256 i; i < networksArray.length; i++) {
+            if (!compareStrings(networksArray[i], domain)) {
+                replicaBeacon = address(
+                    replicaOfUpgrade(domain, networksArray[i]).beacon
+                );
+                break;
+            }
+            homeImpl = address(homeUpgrade(domain).beacon);
+        }
+        homeBeacon = address(homeUpgrade(domain).beacon);
+    }
+
+    function loadImplementations(string memory domain) internal {
+        bridgeTokenImpl = bridgeTokenUpgrade(domain).implementation;
+        tokenRegistryImpl = tokenRegistryUpgrade(domain).implementation;
+        bridgeRouterImpl = bridgeRouterUpgrade(domain).implementation;
+        governanceRouterImpl = governanceRouterUpgrade(domain).implementation;
+        // All replicas have the same implmentation
+        for (uint256 i; i < networksArray.length; i++) {
+            if (!compareStrings(networksArray[i], domain)) {
+                replicaImpl = replicaOfUpgrade(domain, networksArray[i])
+                    .implementation;
+                break;
+            }
+            homeImpl = homeUpgrade(domain).implementation;
+        }
+    }
+
+    function loadController(string memory domain) internal {
+        beaconController = address(getUpgradeBeaconController(domain));
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                       GOVERNANCE CALL GENERATORS
+    //////////////////////////////////////////////////////////////*/
+
+    function generateGovernanceCalls(string memory domain) internal {
+        title("BeaconController upgrade encoded calls for", domain);
+        console2.log(
+            "Domain Number: ",
+            vm.toString(uint256(getDomainNumber(domain)))
+        );
+        console2.log("Function signature: upgrade(address, address)");
+        console2.log(
+            "Arguments: <contract_beacon>, <new_implementation_address>"
+        );
+
+        upgradeHome = abi.encodeWithSignature(
+            "upgrade(address,address)",
```

Comment:
`encodeWithSignature` is generally less preferable to `encodeWithSelector` as the latter is less brittle to interface changes